python/plugins/processing/gui/wrappers.py

```diff
@@ -936,7 +936,10 @@ def createWidget(self):
             self.context = dataobjects.createContext()
 
             try:
-                self.combo.setLayer(iface.activeLayer())
+                if self.parameterDefinition().flags() & QgsProcessingParameterDefinition.FlagOptional:
+                    self.combo.setLayer(self.parameterDefinition().defaultValue())
+                else:
+                    self.combo.setLayer(iface.activeLayer())
```

Comment:
Should the default value also have precedence here and activeLayer only be used as fallback?

Reply:
Maybe you are right, although I have never seen a any usage of default value for layer parameters.

Should be fixed now.

Reply:
I did use it recently, but now I wonder how that even worked (given that the old code was forcing `iface.activeLayer()`)